Disable message signing when Commenting on a Task

rdig opened the pull request:
## Description

This PR disables _(just disables, the underlying message signing system is still there and perfectly functional)_ signing every single comment message before posting it on a task.

This makes the task comment system more fluid and easy to use.

**Changes** 

- [x] removed message signing from the `taskCommentAdd` saga

**Demo** 

![demo-no-signature-comments](https://user-images.githubusercontent.com/1193222/72726909-3e9d9480-3b92-11ea-872b-fbe6baaf4313.gif)

Resolves #1954

ceolson01 commented on src/modules/dashboard/sagas/task.ts:
```diff
@@ -196,10 +196,6 @@ function* taskCommentAdd({
-    yield call(signMessage, 'taskComment', {
-      comment,
-      author,
-    });
```
I know there's a chance that we'll bring this feature back at some point, but I think we can remove this saga (for now at least), no? And just handle the mutation directly in the component?

chmanie replied:
Yes! When we bring it back it won't be handled in a saga

rdig replied:
Good point. I'll make the change

rdig commented:
@chmanie @ceolson01 

Refactored this:
- Removed the task add saga
- Moved the mutation directly in the component

But this introduced a bad, but expected behaviour in the task feed (where comments are displayed).

Because we currently use a poll interval and not a subscription, it takes time, from the time you submit the comment for it to actually show up. 

As you can see in the demo below:

![demo-task-feed-poll-interval](https://user-images.githubusercontent.com/1193222/72794018-f5a61880-3c43-11ea-8f93-4223a3986b31.gif)

Again, this is expected, and will be solved in the future when we introduce subscriptions on the server.

But until then I expect a number of users to complain about this. _(I guess we'll just have to deal with that)_

On the other hand, there's a trick we can use here: a visual one. Delay the the 'isSubmitting' flag by about 4 seconds, meaning it will "take longer" (in the eyes of the users) to submit a comment. And by the time the flag changes to `false`, the task feed would have already been updated.

But I would feel iffy implementing it like that to be honest...

Thoughts ? 

Also, please give it another review

rdig commented:
After taking @chmanie 's advice, I've added in a `refetchQueries` option to the task comment mutation, and now the comment system works properly again. 

Thanks for that Chris!

![demo-comment-refreshed-feed](https://user-images.githubusercontent.com/1193222/72802545-b3d19e00-3c54-11ea-80a3-d25aee5fa468.gif)

@ceolson01 @chmanie this is ready for another review then

ceolson01 commented on src/modules/dashboard/components/TaskComments/TaskComments.tsx:
```diff
@@ -72,26 +76,27 @@ const handleKeyboardSubmit = (
+        refetchQueries: [
+          {
+            query: TaskFeedEventsDocument,
+            variables: { id: draftId, },
```
The type of `variables` isn't inferred from the value of the `query` property, so we should probably use some type assertion here:

```suggestion
            variables: { id: draftId, } as TaskFeedEventsQueryVariables,
```

ceolson01 commented on src/modules/dashboard/components/TaskComments/TaskComments.tsx:
```diff
@@ -72,26 +76,27 @@ const handleKeyboardSubmit = (
+
+  const [sendComment] = useSendTaskMessageMutation();
+  const onSubmit = useCallback(
+    ({ comment: message }: FormValues, { resetForm }: FormikBag<any, any>) =>
```
I think this can be a little more tightly typed:

```suggestion
    ({ comment: message }: FormValues, { resetForm }: FormikBag<object, FormValues>) =>
```
